util/types/time.go

```diff
@@ -1920,7 +1972,81 @@ func minutesNumeric(t *mysqlTime, input string) (string, bool) {
 	return input[2:], true
 }
 
-func dayOfMonthNumericTwoDigits(t *mysqlTime, input string) (string, bool) {
+func time12Hour(t *mysqlTimeEx, input string) (string, bool) {
+	// hh:mm:ss AM
+	if len(input) < 10 {
```

Comment:
you don't take whitespace count into consideration before `AM`
it maybe `hh:mm:ss              AM` or `hh:mm:ssAM`